src/Host/Controllers/Identity/UsersController.cs

```diff
@@ -54,6 +54,19 @@ public Task<string> CreateAsync(CreateUserRequest request)
         return _userService.CreateAsync(request, GetOriginFromRequest());
     }
 
+    [HttpPost("self-register")]
```

Comment:
do we really need another API endpoint for this?

Reply:
the other option is to have a new parameter - leave that decision to you...

Reply:
i was conscious of the importance of keeping them separate in case we need to different logic in the future.

Maybe we should just drop the idea of a new user adding themselves altogether?

 But I also dislike the idea of an admin setting a users password - even in a temporary manner.  

It would perhaps be better to send a temp one-time-code or similar?

Reply:
I think the cleanest is to have an additional endpoint, as you don't want the captcha verification for the regular endpoint. You also don't want to have to supply the TenantId for the regular endpoint... 
It doesn't need to call another method on the service though... can definitely use the same one... the additional checks can happen in the controller endpoint itself. (Maybe captcha verification can also be added via an attribute?).